### ISSUE TITLE
Verify use of `i64` versus `u64` in API

### ISSUE THREAD

**Issue by laanwj** [laanwj/rust-clightning-rpc#10]
Neither the `lightning-cli help` nor the c-lightning man-pages describe the exact type of values and amounts used in requests/responses.

In case it was unknown I have used `i64` everywhere, except in one case where it caused problems (`max_htlc_value_in_flight_msat` in `listpeers`). However, it would be good to go over the structures in `requests` and `responses` and reconcile them with the C source code.

**Comment by Kixunil** [comment]
I'd personally prefer newtype pattern in this case. We're dealing with money and it might be easy to confuse satoshis vs millisatoshis.

Regarding signedness, it makes more sense to me to use `u64`, since paying negative amount makes little sense to me, but I can imagine `i64` being used in some calculations.

**Comment by laanwj** [comment]
My point here is that the types in the requests/responses simply have to match the parsing/serialization on the server side. This prevents parse errors at either side.

Not all the numbers on the interface are amounts of money, FWIW. Some things can have negative values!

For the high-level interface it'd be an idea to use some more specific wrapper type, sure.

**Comment by Kixunil** [comment]
Yeah, I meant it more generally - if some number shouldn't be negative, use unsigned number.

Of course, the types should match. If the API uses e.g. signed number for amounts, I'd consider it a bug (or imperfection, maybe) on the server side.

**Comment by laanwj** [comment]
On the generation side, there are two relevant calls.
```C
/* '"fieldname" : value' or 'value' if fieldname is NULL */
void json_add_num(struct json_result *result, const char *fieldname,
                  unsigned int value);
/* '"fieldname" : value' or 'value' if fieldname is NULL */
void json_add_u64(struct json_result *result, const char *fieldname,
                  uint64_t value);
```
On the parsing side
```C
/* Extract number from this (may be a string, or a number literal) */
bool json_to_number(const char *buffer, const jsmntok_t *tok,
                    unsigned int *num);
/* Extract number from this (may be a string, or a number literal) */
bool json_to_u64(const char *buffer, const jsmntok_t *tok,
                 uint64_t *num);
```
There's nothing to suggest signed values at all, so it looks like you're right.
